lib/process-services-cloud/src/lib/people/components/people-cloud.component.ts

```diff
@@ -316,15 +316,22 @@ export class PeopleCloudComponent implements OnInit, OnChanges, OnDestroy {
         this.isLoading = false;
     }
 
-    isPreselectedUserValid(preselectedUser: IdentityUserModel, validatedUser: IdentityUserModel) {
-        if (validatedUser && (validatedUser.id !== undefined || validatedUser.username !== undefined || validatedUser.email !== undefined)) {
-            return preselectedUser.id === validatedUser.id || preselectedUser.username === validatedUser.username || preselectedUser.email === validatedUser.email;
-        } else {
-            return false;
+    isPreselectUserAlignedWithValidatedUser(preselectedUser: IdentityUserModel, validatedUser: IdentityUserModel) {
+        const key = this.getUserSearchKey(preselectedUser);
+
+        switch (key) {
```

Comment:
I think some abstraction is missing from here, this file contains switch/case structures at least 3 times for the same routing. This is a scholar example of violation the OCP. Please have a look at on this article for more information:
https://stackify.com/solid-design-open-closed-principle/

Reply:
Do wen even need this?
I mean something like `return preselectedUser[key] === validatedUser[key]` can't work?

Reply:
Totally agree. A maintenance ticket for the next release has been created as there is no time to be refactored in this release. https://issues.alfresco.com/jira/browse/AAE-1652